lib/chef/resource/hostname.rb

```diff
@@ -270,13 +270,24 @@ def updated_ec2_config_xml
 
           unless Socket.gethostbyname(Socket.gethostname).first == new_resource.hostname
             converge_by "set hostname to #{new_resource.hostname}" do
+              puts "What username is being used : #{new_resource.domain_user}"
               powershell_exec! <<~EOH
-                if ([string]::IsNullOrEmpty(#{new_resource.domain_user})){
+                $domain_joined = (Get-WmiObject -Class Win32_ComputerSystem).PartofDomain
+
+                if (-not $domain_joined){
                   Rename-Computer -NewName #{new_resource.hostname}
                 }
                 else {
-                  $user = #{new_resource.domain_user}
-                  $secure_password = #{new_resource.domain_password} | Convertto-SecureString -AsPlainText -Force
+                  $temp_user = #{new_resource.domain_user}
+                  $temp_password = #{new_resource.domain_password}
+                  if ([string]::IsNullOrEmpty($temp_user)){
+                    $temp_user = "Chef"
+                  }
+                  if ([string]::IsNullOrEmpty($temp_password)){
+                    $temp_user = "P@ssw0rd"
+                  }
```

Comment:
will this `chef` credential work if it does not exist? Seems like you at least need a comment here explaining.